src/main/java/frc/robot/subsystems/arm/ArmIOReal.java

```diff
@@ -123,235 +144,205 @@ public void setShoulderJointPower(double power) {
+             * Sets the position of the end of the arm.
+             *
+             * @param armLocation Translation2d of the desired location.
+             */
+            public void setEndPosition (Translation2d armLocation){
+                setEndPosition(armLocation, 0, 0);
+            }
+
+            /**
+             * Sets the position of the end of the arm.
+             *
+             * @param armLocation Translation2d of the desired location.
+             */
+            public void setEndPosition (Translation2d armLocation,double shoulderFFMultiplier, double elbowFFMultiplier)
+            {
+                var angles = kinematics.inverseKinematics(armLocation);
+                setShoulderJointAngle(Math.toDegrees(angles.shoulderAngle), shoulderFFMultiplier, 0);
+                setElbowJointAngle(Math.toDegrees(angles.elbowAngle), elbowFFMultiplier, 0);
+                ySetpoint = armLocation.getY();
+            }
+
+            /**
+             * Gets the velocity of the shoulder motors.
+             *
+             * @return shoulder motors velocity. [rad/sec]
+             */
+            public double getShoulderMotorVelocity () {
+                return shoulderMainMotor.getVelocity().getValue();
+            }
+
+            public void setFinalSetpointAngles (Translation2d position, ArmInputsLogged inputs){
+                var solution = kinematics.inverseKinematics(position);
+                inputs.finalSetpointAngles[0] = Math.toDegrees(solution.shoulderAngle);
+                inputs.finalSetpointAngles[1] = Math.toDegrees(solution.elbowAngle);
+            }
+
+            /**
+             * Gets the velocity of the elbow motors.
+             *
+             * @return elbow motor velocity. [rad/sec]
+             */
+            public double getElbowMotorVelocity () {
+                return elbowMainMotor.getVelocity().getValue();
+            }
+
+            public void resetArmEncoders () {
+                double elbowAngleReset = elbowEncoder.getAbsolutePosition() * 360.0 - ArmConstants.ELBOW_ZERO_POSITION;
+                double shoulderAngleReset = shoulderEncoder.getAbsolutePosition() * 360.0 - ArmConstants.SHOULDER_ZERO_POSITION;
+                shoulderOffset = AngleUtil.normalize(shoulderAngleReset) / 360.0;
+                elbowOffset = AngleUtil.normalize(elbowAngleReset) / 360.0;
+            }
+
+            public Translation2d getElbowJointPosition () {
+                Rotation2d shoulderAngle = getShoulderJointAngle();
+                return new Translation2d(
+                        ArmConstants.SHOULDER_ARM_LENGTH * shoulderAngle.getCos(),
+                        ArmConstants.SHOULDER_ARM_LENGTH * shoulderAngle.getSin());
+            }
+
+            public boolean armIsOutOfFrame () {
+                Translation2d elbowJoint = getElbowJointPosition(), endPosition = getEndPosition();
+                return !(elbowJoint.getX() < 0) || !(endPosition.getX() < 0);
```

Comment:
```suggestion
                return elbowJoint.getX() >= 0 || endPosition.getX() >= 0;
```
But is that true, though? Shouldn't positive values indicate "in frame"?